allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -39,9 +50,113 @@ def __init__(self, cache_dir, project_name):
 
         self._manifest = None
         self._cache_dir = cache_dir
+
         self._project_name = project_name
         self._manifest_file_names = self._list_all_manifests()
 
+        # what latest_manifest was the last time an OutdatedManifestWarning
+        # was emitted
+        self._manifest_last_warned_on = None
+
+        # self._downloaded_data_path is where we will keep a JSONized
+        # dict mapping paths to downloaded files to their file_hashes;
+        # this will be used when determining if a downloaded file
+        # can instead be a symlink
+        c_path = pathlib.Path(self._cache_dir)
+        self._downloaded_data_path = c_path / '_downloaded_data.json'
+
+        # if the local manifest is missing but there are
+        # data files in cache_dir, emit a warning
+        # suggesting that the user run
+        # self.construct_local_manifest
+        if not self._downloaded_data_path.exists():
+            file_list = c_path.glob('**/*')
+            has_files = False
+            for fname in file_list:
+                if fname.is_file():
+                    if 'json' not in fname.name:
+                        has_files = True
+                        break
+            if has_files:
+                msg = 'This cache directory appears to '
+                msg += 'contain data files, but it has no '
+                msg += 'record of what those files are. '
+                msg += 'You might want to consider running\n\n'
+                msg += 'self.construct_local_manifest()\n\n'
```

Comment:
I think mention of `self` may end up being confusing for end users. Maybe instead saying something like:

`You might want to consider running f"{type(self).__name__}.construct_local_manifest()"`

Reply:
Do you think that will be clear enough for users interacting with this principally through the BehaviorOphysProjectCache?

Should we add aliases to BehaviorOphysProjectCache for `construct_local_manifest` and `compare_manifests` so that users can call those functions without having to find the underlying S3CloudCache?

Reply:
Aliases to BehaviorOphysProjectCache would probably be clearer. In addition to the ones you mention `load_latest_manifest()` and `load_manifest()` would probably also be good to expose. Maybe there should be a `show_current_manifest()` property so that users will know what they are currently using?

Reply:
Aliases have been added to VisualBehaviorOphysProjectCache